### PR TITLE
Quarkus

### PR DESCRIPTION
**Pull request by denis-anisimov**
## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes # (issue)

## Type of change

- [ ] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+A base starter project https://github.com/vaadin/base-starter-flow-quarkus/ can be used
+as a starting point for your own application.
+
+== Vaadin CDI Features
+
+Quarkus is a CDI container so it's possible to use all CDI features.
+Check the CDI documentation for these features:
+- <<../cdi/contexts#, Vaadin CDI Scopes>> 
+- <<../cdi/events#, Observable Vaadin Eventss>>
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:26]:
```suggestion
- <<../cdi/events#, Observable Vaadin Events>> 
```

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+are required.
+The Quarkus application can be run using `quarkus-maven-plugin` using `quarkus:dev` goal.
+
+A base starter project https://github.com/vaadin/base-starter-flow-quarkus/ can be used
+as a starting point for your own application.
+
+== Vaadin CDI Features
+
+Quarkus is a CDI container so it's possible to use all CDI features.
+Check the CDI documentation for these features:
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:24]:
```suggestion
See the documentation for Vaadin CDI features:
```

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+- <<../cdi/contexts#, Vaadin CDI Scopes>> 
+- <<../cdi/events#, Observable Vaadin Eventss>> 
+- <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>
+- <<../cdi/nstantiated-beans#, Using CDI Beans>>
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:28]:
```suggestion
- <<../cdi/instantiated-beans#, Using CDI Beans>>
```

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+[[quarkus.vaadin.addons]]
+= Vaadin add-ons in Quarkus Application
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:32]:
```suggestion
== Vaadin add-ons In Quarkus Applications
```
Deployment seems to not accept main headers inside the text...

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+The Vaadin add-on used in Quarkus should have jandex index. This can be done via `jandex-maven-plugin`.
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:34]:
```suggestion
Any Vaadin add-on used in a Quarkus application should contain a jandex index.
This is generated using the `jandex-maven-plugin`.
See https://quarkus.io/guides/cdi-reference#how-to-generate-a-jandex-index[How to Generate a Jandex Index]
```

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+If you can’t modify the dependency, you can still index it by adding `quarkus.index-dependency` 
+entries to your application.properties:
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:36]:
```suggestion
If you can’t modify the dependency, you can still get Quarkus to index it by adding `quarkus.index-dependency` 
entries to your application.properties:
```

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+quarkus.index-dependency.<name>.artifact-id=
+quarkus.index-dependency.<name>.classifier=(this one is optional)
+----
+
+
+[[quarkus.vaadin.limitations]]
+== Limitations
+
+
+Websockets Push doesn't work currently in Vaadin integration for Quarkus.
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:51]:
```suggestion
WebSockets Push doesn't work currently in the Vaadin extension for Quarkus.
```

**Review by caalador — CHANGES_REQUESTED** [review]
Also I think add `Quarkus` and `CDI` to .github/styles/Vocab/Docs/accept.txt
to not get the nagging `did you really mean`

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+[[quarkus.basic]]
+= Using Quarkus with Vaadin
+
+Quarkus is an open source, Kubernetes-native Java framework that allows 
+developers to optimize container-native applications for peak performance.
+Check https://quarkus.io/ to get more info about Quarkus.
+
+To be able to run Quarkus  the `vaadin-quarkus` and `vaadin-core-jandex` dependencies
```
[the remark this answers is not in the document]
Comment (denis-anisimov) [reply articles/flow/integrations/quarkus/basic.asciidoc]: I have no idea

I've just checked the base starter structure and put here the vaadin plugins which I found there.
If you say that it's not needed let's remove it.
Should I just remove it or replace with something ?

**Comment by denis-anisimov** [comment]
> Also I think add `Quarkus` and `CDI` to .github/styles/Vocab/Docs/accept.txt
> to not get the nagging `did you really mean`

done

File: articles/flow/integrations/quarkus/basic.asciidoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (caalador) [reply articles/flow/integrations/quarkus/basic.asciidoc]: well at the moment it is needed as I don't think it's part of the vaadin(-core) dependencies, but perhaps we should just say vaadin-jandex as it works with both (vaadin/vaadin-core) cases and doesn't have an impact on the size of the compiled project.

[the remark this answers is not in the document]
Comment (denis-anisimov) [reply articles/flow/integrations/quarkus/basic.asciidoc]: done

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+as a starting point for your own application.
+
+== Vaadin CDI Features
+
+Quarkus’ dependency injection solution is based on CDI, so it's possible to use all CDI features.
+See the documentation for Vaadin CDI features:
+- <<../cdi/contexts#, Vaadin CDI Scopes>> 
+- <<../cdi/events#, Observable Vaadin Events>> 
+- <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>
+- <<../cdi/instantiated-beans#, Using CDI Beans>>
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:27]:
```suggestion
* <<../cdi/contexts#, Vaadin CDI Scopes>> 
* <<../cdi/events#, Observable Vaadin Events>> 
* <<../cdi/service-beans#, Vaadin Service Interfaces as CDI Beans>>
* <<../cdi/instantiated-beans#, Using CDI Beans>>
```
I guess these should be with `*`s as they are not getting a new line in the preview: https://deploy-preview-737--vaadin-docs-preview.netlify.app/flow/integrations/quarkus/basic/#vaadin-cdi-features

**Review by caalador — DISMISSED** [review]
`jandex` `Jandex` could also be in the accepted file.
Then just the list issue so that they wouldn't be after each other withouth new lines.

**Comment by denis-anisimov** [comment]
> `jandex` `Jandex` could also be in the accepted file.
> Then just the list issue so that they wouldn't be after each other withouth new lines.

done

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+Using the `quarkus-maven-plugin` the `quarkus:dev` goal is used to start the application.
+
+The https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] project  can be used
+as a starting point for your own application.
+
+== Vaadin CDI Features
+
+Quarkus’ dependency injection solution is based on CDI, so it's possible to use all CDI features.
+See the documentation for Vaadin CDI features:
+* <<../cdi/contexts#, Vaadin CDI Scopes>>
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:24]:
```suggestion
See the documentation for Vaadin CDI features:

* <<../cdi/contexts#, Vaadin CDI Scopes>> 
```
Need new line else seen as one line?

File: articles/flow/integrations/quarkus/basic.asciidoc [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+Quarkus is an open source, Kubernetes-native Java framework made for Java virtual machines and native compilation, optimizing Java specifically for containers and enabling it to become an effective platform for serverless, cloud, and Kubernetes environments.
+Check https://quarkus.io/ to get more info about Quarkus.
+
+To be able to run Quarkus the `vaadin-quarkus` and `vaadin-jandex` dependencies 
+are required.
+Using the `quarkus-maven-plugin` the `quarkus:dev` goal is used to start the application.
+
+The https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] project  can be used
+as a starting point for your own application.
```
Comment (caalador) [inline articles/flow/integrations/quarkus/basic.asciidoc:18]:
```suggestion
The https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] project can be used
as a starting point for your own application.
There is also the https://github.com/vaadin/base-starter-flow-quarkus/tree/gradle[base starter for gradle].
```

**Review by magi42 — APPROVED** [review]

**Pull request merged**